### PR TITLE
Iteration 2 tt

### PR DESCRIPTION
**Pull request by tylertomlinson**
Iteration 2 complete.
All tests passing using Rakefile / 100% coverage with Simplecov.

 Using helper methods to validate placement of ships:
appropriate_length?
ships_cannont_overlap?
consecutive --- using letter_coordinates method and number_coordinates method to find consecutive_letter_coordinates as well as consecutive_number_coordinates. Then if either of those is true within the consecutive method the return value is true.

     Each method I put into valid_placement? and if all are true the return value is true.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    coordinates.map(&:chars).map {|coordinate| coordinate[1].to_i}
+  end
+
+  def consecutive_letter_coordinates?(coordinates)
+    number_coordinates(coordinates).uniq.size == 1 &&
+    letter_coordinates(coordinates).map { |letter| letter.ord }.each_cons(2).all? { |number1, number2| number2 == (number1 + 1) }
+  end
+
+  def consecutive_number_coordinates?(coordinates)
+    letter_coordinates(coordinates).uniq.size == 1 &&
```
Comment (chrisstankus99) [inline lib/board.rb:59]: Where is this being tested?

File: test/board_test.rb [hunk 2]
```diff
@@ -11,30 +11,61 @@ def setup
...
+    refute @board.valid_placement?(@cruiser, ["A1", "B2", "C3"])
+    refute @board.valid_placement?(@submarine, ["C2", "D3"])
+  end
+
+  def test_ship_is_able_to_be_placed_on_board_and_can_occupy_multiple_cells
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+
+    cell_1 = @board.cells["A1"]
+    cell_2 = @board.cells["A2"]
+    cell_3 = @board.cells["A3"]
```
Comment (chrisstankus99) [inline test/board_test.rb:65]: Why are these not instance variables?(Just curious for my own knowledge)

**Comment by chrisstankus99** [comment]
Nice job on the helper methods and boolean logic in this.

File: lib/board.rb (hunk 1 above)
Comment (tylertomlinson) [reply lib/board.rb:59]: All the methods(helper methods) are within valid_placement? so they are being tested within the three valid placement tests. I had broken each out initially then combined as I refactored.

File: test/board_test.rb (hunk 2 above)
Comment (tylertomlinson) [reply test/board_test.rb:65]: We are only using them within this test if I remember correctly. WE MUST PROTECT OUR CODE!

**Pull request merged**